### PR TITLE
Add analytics to the command autocomplete

### PR DESCRIPTION
**Pull request by iomodo**
#### Summary
This PR will add analytics for slash command autocomplete

Testing notes: verify that analytics data being tracked is correct for commands and previous collected data types.

#### Ticket Link
  Fixes https://mattermost.atlassian.net/browse/MM-27134
#### Device Information
This PR was tested on: iphone 11, OS 13.5(simulator)

#### Screenshots
No UI changes

### CODE REVIEW

**Comment by aaronrothschild** [comment]
@iomodo Does this track only on mobile, or was only tested on mobile?

File: app/mm-redux/utils/helpers.ts [hunk 1]
```diff
@@ -121,3 +121,21 @@ export function buildQueryString(parameters: Dictionary<any>): string {
 
     return query;
 }
+
+export function sanitizeCommandForAutocompleteTracking(userInput: string): string {
+    const commandList = ['agenda', 'autolink', 'away', 'bot-server', 'code', 'collapse',
```
Comment (aaronrothschild) [inline app/mm-redux/utils/helpers.ts:126]: @iomodo Just curious - are these the commands that are "whitelisted" for autocompletion?  Do we need to add anything in the future if say, a custom slash command was added?

**Comment by aaronrothschild** [comment]
FYI @amyblais this is the telemetry PR I was talking about.

File: app/mm-redux/utils/helpers.ts (hunk 1 above)
Comment (iomodo) [reply app/mm-redux/utils/helpers.ts:126]: @aaronrothschild The tracking is only for mobile. We need to understand whether we need caching or not right?
These are all the slash commands we have on the community. Bunch of them does not make sense for mobile(for example `shortcuts`), but I still included them just to track if someone is trying to use them on mobile.

**Comment by DHaussermann** [comment]
Alex has shared steps with me on how telemetry can be tested on an individual plugin build but, I'm not sure that's applicable here. 
@Adovenmuehle or @iomodo, can you share any guidance on if this is testable before merge? Is it simply a matter of deploying multiple plugins built to support telemetry and using mobile auto-complete or, is there something else needed here?

**Comment by Adovenmuehle** [comment]
@DHaussermann oh I didn't realize you were asking about this specifically. Since this is just in mobile, if we can have a CI build kicked off, then it will automatically be sending data to Rudder/Looker.

**Comment by DHaussermann** [comment]
No worries @alexrashed I was asking a broader question at the time :)  

Let me see if CI will build this for us now.

**Comment by DHaussermann** [comment]
No worries @Adovenmuehle I was asking a broader question at the time as there are a couple other plugin specific PRs :)

Let me see if CI will build this for us now.

**Comment by DHaussermann** [comment]
@Adovenmuehle quick follow-up on this - Can you confirm one of the servers this build ☝️  can be connected to so I can generate data visible in looker? On looker, I see many server IDs but not sure what URL those servers those servers are at.

I realize this still requires dev review but I want to try and pin down the testing process.

**Comment by DHaussermann** [comment]
@iomodo Not sure if there was a list of what we want to capture here. I see `complete_suggestion`, `get_suggestions_initiated` and `get_suggestions_success`
Autocomplete-data

The data looks accurate to me. Do you know anything else we should be capturing?

**Comment by iomodo** [comment]
@DHaussermann we are capturing failures as well, when network is down or you have no permission or whatever the case may be.

**Comment by DHaussermann** [comment]
@iomodo, one last question on this. How does the get suggestion work in the event that there is no network access? Does it continue to attempt to fetch suggestions repeatedly? 

I ask because the count seems to exceed the amount of times I failed to a suggestion. 
  
Rotter

I tried this a few times with my device in airplane mode to load them (about 5) but, then saw 31 occurrences. Not sure if this is concenring or perhaps simply a matter of something else I was doing also caused this event. 

This may be beyond the scope of this PR, thoughts?

**Comment by iomodo** [comment]
@DHaussermann that probably is the expected behavior. The request is fired every time you change the input text so when you type something or delete something you will get `get_suggestions_initiated` event and either `get_suggestions_failed` or `get_suggestions_success`.
There also should be the commands which were inputed and in case of the error we are sending the error message to understand what type of errors happen. Cana you confirm that this informations is also there?

**Comment by DHaussermann** [comment]
@iomodo there is no unique event that captures errors I can see. But, on the failed events an error does get captured. For example 
`"ERROR": "Received invalid response from the server."`
Is this what you mean?

**Comment by iomodo** [comment]
@DHaussermann Yes that's exactly what I meant. I suspect it will give us enough information.

**Pull request merged**